File: v6/prolog/compile/test/typegen_golden/compiler-derived-relation.types.ts
export type Option<T> = { tag: 'none' } | { tag: 'some'; value: T };

export interface Holder {
  value: GenPartialUser9d7a703929b72789;
}

export interface User {
  id: number;
  name: string;
}

export interface GenPartialUser9d7a703929b72789 {
  id: Option<number>;
  name: Option<string>;
}

File: v6/prolog/compile/test/typegen_golden/compiler-derived-relation.types.rs
#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
#[serde(tag = "tag", content = "value", rename_all = "snake_case")]
pub enum DlOption<T> {
    None,
    Some(T),
}

#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
pub struct Holder {
    pub value: GenPartialUser9d7a703929b72789,
}

#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
pub struct User {
    pub id: i64,
    pub name: String,
}

#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
pub struct GenPartialUser9d7a703929b72789 {
    pub id: DlOption<i64>,
    pub name: DlOption<String>,
}
